### ISSUE TITLE
Separator Block: Color is not reflected correctly

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#41194]
### Description

The color of the separator blocks is not reflected correctly in both the Classic and Block themes.

I think that this problem might be caused by the fact that the color setting has been moved to block support and the `.has-background-xxx` class is no longer assigned in #38428.

### Step-by-step reproduction instructions

1. Activate Twenty Twenty Two (Block theme).
2. Confirm that solid colors are reflected correctly, but gradient colors are not.
3. Activate Twenty Twenty Two (Classic theme).
4. Confirm that both solid and gradient colors are not reflected.

## Block Theme (Twenty Twenty Theme)

https://user-images.githubusercontent.com/54422211/169508446-848e0eea-7a30-425c-96fe-8680511e4f33.mp4

## Classic Theme (Twenty Twenty One)

https://user-images.githubusercontent.com/54422211/169508470-8971b49a-391c-4a47-9712-3ef20c48f59b.mp4

### Environment info

WordPress Version: 6.0-RC3
Gutenberg Version: 13.2.2
Theme: Twenty Twenty Two / Twenty Twenty One

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by carolinan** [comment]
Yes I would consider the removal of the class  is a regression, since theme authors has relied on this class.
I did not understand the logic behind keeping the text color class, when the setting we are changing is called background color.
